frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareAppStoreVpp/SoftwareVppForm/SoftwareVppForm.tsx

```diff
@@ -0,0 +1,318 @@
+
+import { generateFormValidation, getUniqueAppId } from "./helpers";
+import { CUSTOM_TARGET_OPTIONS } from "../../SoftwareFleetMaintained/FleetMaintainedAppDetailsPage/FleetAppDetailsForm/helpers";
+
+const baseClass = "software-vpp-form";
+
+const NoVppAppsCard = () => (
+  <Card borderRadiusSize="medium" paddingSize="xxxlarge">
+    <div className={`${baseClass}__no-software-message`}>
+      <p className={`${baseClass}__no-software-title`}>
+        You don&apos;t have any App Store apps
+      </p>
+      <p className={`${baseClass}__no-software-description`}>
+        You must purchase apps in ABM. App Store apps that are already added to
+        this team are not listed.
+      </p>
+    </div>
+  </Card>
+);
+
+interface IVppAppListItemProps {
+  app: IVppApp;
+  selected: boolean;
+  uniqueAppId: string;
+  onSelect: (software: IVppApp) => void;
+}
+
+const VppAppListItem = ({
+  app,
+  selected,
+  uniqueAppId,
+  onSelect,
+}: IVppAppListItemProps) => {
+  return (
+    <li className={`${baseClass}__list-item`}>
+      <Radio
+        label={
+          <div className={`${baseClass}__app-info`}>
+            <SoftwareIcon url={app.icon_url} />
+            <span>{app.name}</span>
+          </div>
+        }
+        id={`vppApp-${uniqueAppId}`}
+        checked={selected}
+        value={uniqueAppId}
+        name="vppApp"
+        onChange={() => onSelect(app)}
+      />
+      {app.platform && (
+        <div className="app-platform">
+          {PLATFORM_DISPLAY_NAMES[app.platform]}
+        </div>
+      )}
+    </li>
+  );
+};
+
+interface IVppAppListProps {
+  apps: IVppApp[];
+  selectedApp: IVppApp | null;
+  onSelect: (app: IVppApp) => void;
+}
+
+const VppAppList = ({ apps, selectedApp, onSelect }: IVppAppListProps) => {
+  const uniqueSelectedAppId = selectedApp ? getUniqueAppId(selectedApp) : null;
+  return (
+    <div className={`${baseClass}__list-container`}>
+      <ul className={`${baseClass}__list`}>
+        {apps.map((app) => {
+          const uniqueAppId = getUniqueAppId(app);
+          return (
+            <VppAppListItem
+              key={uniqueAppId}
+              app={app}
+              selected={uniqueSelectedAppId === uniqueAppId}
+              uniqueAppId={uniqueAppId}
+              onSelect={onSelect}
+            />
+          );
+        })}
+      </ul>
+    </div>
+  );
+};
+
+export interface ISoftwareVppFormData {
+  selfService: boolean;
+  targetType: string;
+  customTarget: string;
+  labelTargets: Record<string, boolean>;
+  selectedApp?: IVppApp | null;
+}
+
+export interface IFormValidation {
+  isValid: boolean;
+  customTarget?: { isValid: boolean };
+}
+
+interface ISoftwareVppFormProps<T extends "add" | "edit"> {
+  labels: ILabelSummary[] | null;
+  vppApps?: T extends "add" ? IVppApp[] : never;
+  softwareVppForEdit?: T extends "edit" ? IAppStoreApp : never;
+  onSubmit: (formData: ISoftwareVppFormData) => void;
+  isUploading?: boolean;
```

Comment:
Can we call this `isLoading` without breaking all the things/creating an inconsistent interface? We don't upload anything when we add a VPP app, unlike with FMA/custom packages.